mevislab.github.io/content/tutorials/thirdparty/matplotlib/3Dplotting.md

```diff
@@ -20,6 +20,49 @@ In this tutorial, we will equip the macro module we created in the [Example 1: M
 
 ## Steps to do 
 
-The fields and commands needed have already been prepared in the second tutorial. We will just have to modify our .py file a little to make them usable.
+The fields and commands needed have already been prepared in the second tutorial. We will just have to modify our .py file a little to make them usable. Integrate the following code into your .py file:
 
+{{< highlight filename = "BaseNetwork.py">}}
+```Stan
+def click3D():
+  clearFigure()
+  figure = ctx.control("canvas").object().figure()
+  
+  global values
+  values = [i for i in range(startSlice, lastSlice)]
+  
+  if startSlice == endSlice:
+    subplot = figure.add_subplot(111, projection='3d')
+    z = 0
+    dx = 1
+    dy = 1
+    subplot.bar3d(getX(),startSlice,z,dx,dy,getY())
```

Comment:
Instead of declaring these as separate variables, you could just use named arguments to make it clear what the values are for (same below).